### PR TITLE
Add pagination to issue fetching

### PR DESCRIPTION
**Pull request by mason-at-pieces**

### CODE REVIEW

**Review by caleb-at-pieces — COMMENTED** [review]
would really like some JS DOC comments so we can clearly define what the param and return types are

File: .github/stale-issue-checker/index.js [hunk 1]
```diff
@@ -1,15 +1,64 @@
...
+          "2022-11-28",
+      },
+    });
+
+    const parsedData = parseData(response.data)
+    data = [...data, ...parsedData];
+
+    const linkHeader = response.headers.link;
+
+    pagesRemaining = linkHeader && linkHeader.includes(`rel=\"next\"`);
```
Comment (caleb-at-pieces) [inline .github/stale-issue-checker/index.js:24]: so this is web scraping? is there not a good way to do this with just the api?

File: .github/stale-issue-checker/index.js [hunk 2]
```diff
@@ -1,15 +1,64 @@
...
+  // Otherwise, the array of items that we want is in an object
+  // Delete keys that don't include the array of items
+  delete data.incomplete_results;
+  delete data.repository_selection;
+  delete data.total_count;
+  // Pull out the array of items
+  const namespaceKey = Object.keys(data)[0];
+  data = data[namespaceKey];
+
+  return data;
```
Comment (caleb-at-pieces) [inline .github/stale-issue-checker/index.js:55]: let's consolidate this to :

```javascript
const namespaceKey = Object.keys(data).find((key) => !['incomplete_results', 'repository_selection', 'total_count'].includes(key))
return data[namespaceKey]
```

File: .github/stale-issue-checker/index.js [hunk 3]
```diff
@@ -1,15 +1,64 @@
...
+    const response = await octokit.request(`GET ${url}`, {
+      per_page: 100,
+      headers: {
+        "X-GitHub-Api-Version":
+          "2022-11-28",
+      },
+    });
+
+    const parsedData = parseData(response.data)
+    data = [...data, ...parsedData];
```
Comment (caleb-at-pieces) [inline .github/stale-issue-checker/index.js:20]:
```javascript
data = data.concat(parseData(response.data));
```

File: .github/stale-issue-checker/index.js (hunk 1 above)
Comment (mason-at-pieces) [reply .github/stale-issue-checker/index.js:24]: This is straight from the docs on how to paginate haha

File: .github/stale-issue-checker/index.js (hunk 3 above)
Comment (mason-at-pieces) [reply .github/stale-issue-checker/index.js:20]: updated

File: .github/stale-issue-checker/index.js (hunk 1 above)
Comment (mason-at-pieces) [reply .github/stale-issue-checker/index.js:23]: Not web scraping, just look into the link headers: https://docs.github.com/en/rest/using-the-rest-api/using-pagination-in-the-rest-api?apiVersion=2022-11-28#using-link-headers

Comment (caleb-at-pieces) [reply .github/stale-issue-checker/index.js:23]: yea i see it's not webscraping just definitely has a webscraping vibe to it

**Review by caleb-at-pieces — APPROVED** [review]
good looks

**Comment by mason-at-pieces** [comment]
> good looks

nothing good ole jetbrains pieces copilot cant do for me haha

**Pull request merged**